### PR TITLE
Merge AnacondaRecipes recipe

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
ping

**Comment by ocefpaf** [comment]
LGTM

**Comment by jakirkham** [comment]
LGTM

File: recipe/build.sh [hunk 1]
```diff
@@ -1,8 +1,9 @@
 #!/bin/bash
 
-CFLAGS="-fPIC" ./configure --shared --prefix=$PREFIX
+./configure --prefix=${PREFIX}  \
```
Comment (pitrou) [inline recipe/build.sh:3]: This change (the removal of `"-fPIC"`) seems to be breaking Arrow builds on Travis-CI:
https://travis-ci.org/pitrou/arrow/jobs/385063783#L2321

@jakirkham

Comment (ocefpaf) [reply recipe/build.sh:3]: I thought that we had `-fPIC` set globally now :-/

**Comment by wesm** [comment]
@ocefpaf how can we help figure out the -fPIC issue? I wonder how we can test that a particular library has been built properly for static linking

cc @cpcloud @kszucs

**Comment by ocefpaf** [comment]
@msarahan or @mingwandroid can probably comment if the `-fPIC` is included or not in the default flags used with the new compilers.

If not I'm :+1: on adding them. For now we could re-add then here and issue a new build number, but I'd rather see this fixed globally if possible.

**Comment by pitrou** [comment]
There is a suggestion on how to test for PICness here (for Linux at least):
https://stackoverflow.com/a/19768349/10194

Perhaps that can be automated.

**Comment by msarahan** [comment]
-fPIC is the default in the new compilers: https://github.com/AnacondaRecipes/aggregate/blob/master/ctng-compilers-activation-feedstock/recipe/conda_build_config.cos6.x86_64.yaml#L42

**Comment by ocefpaf** [comment]
~~@pitrou and @wesm I'm not against testing if that is what you need, but once we define a proper policy for that flag it should be enable and testing is not really needed.~~

> -fPIC is the default in the new compilers: https://github.com/AnacondaRecipes/aggregate/blob/master/ctng-compilers-activation-feedstock/recipe/conda_build_config.cos6.x86_64.yaml#L42

OK. Now I'm lost :confounded:

**Comment by isuruf** [comment]
This build is not using the new compilers. It's using toolchain.

**Comment by ocefpaf** [comment]
Things are getting really hard to follow, see ``{{ compiler('c') }}`` but I'm not seeing toolchain anywhere. Anyway, a short-term fix is to re-add `-fPIC` here and bump the build number.

**Comment by msarahan** [comment]
toolchain right now is the value that gets used for {{ compiler('c') }}.  That jinja2 doesn't mean that the new compilers are used.  It's a syntax that makes it easier to swap out compilers.  @isuruf made a simple wrapper around the old toolchain stuff to get it to work.

The other part of this story is https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L12

That's where toolchain stuff is being specified, overriding the conda-build provided defaults (which would be the new Anaconda compilers.)

The thought is that with all the recipes swapped to the new style, it'll be trivial to switch to the new compilers - just remove that bit in conda-forge-pinning, re-render, and rebuild.

**Comment by jakirkham** [comment]
Indeed we don’t want to be on `toolchain` long term, but switching requires the rebuild to proceed and the rebuilt packages tested. FWIW we have people ready to test, but I think Boost was missing last I checked.

**Comment by msarahan** [comment]
If boost is all that is blocking some good testing, I will work to get that up on the testing channel soon.

**Comment by jakirkham** [comment]
@marcelotrevisani ^ ;)

Thanks @msarahan. That would be very helpful.

**Comment by jakirkham** [comment]
> Anyway, a short-term fix is to re-add `-fPIC` here and bump the build number.

Done in PR ( https://github.com/conda-forge/zlib-feedstock/pull/22 ).

**Comment by msarahan** [comment]
boost packages are being uploaded now.  Please ping me on any issues people see with testing.

**Comment by msarahan** [comment]
I should note - linux and mac only.  Windows has issues, as usual.
